### DISCUSSION TITLE
Issues when using tinyUSB with MAX3421 HCD API

### DISCUSSION THREAD

**Discussion by jvelazquez98** [hathach/tinyusb#2851] [Q&A] [answered]
I'm trying to use tinyUSB together with the already provided max3421-hcd layer on my device. The HW that I'm using connects the MAX3421E USB line to a USB Hub (CYUSB3304), reason why I'm configuring the proper macro on tinyUSB compilation for this purpose. However, when starting the USB enumeration process, I'm always getting a STALLED status, indicating that the hub device is not being able to communicate with the MAX3421E.

I have tried to generate a petalinux image and use the linux USB stack for this same hardware, and it works perfectly, so I assume it should not be a problem on the hub device nor on the hardware configuration.

Given this situation, I would like to know if it is possible to have some examples where the max3421-hcd layer is used, and also some more documentation explaining the usage of this file.

Also, if anyone has any idea why the Hub device is returing the STALL status, I would really appreciate it.

Thanks a lot in advance!

Best regards,
Jaime

**Comment by hathach** [comment, marked as answer]
hcd max3421 does not support hub fornow

**Reply by jvelazquez98** [reply]
Is it expected to be developed in some near future? Are there any guidelines or examples from other devices, on which modifications should be made on this module?

Regards,
Jaime

**Reply by hathach** [reply]
I currently have no plan for adding hub support for max3421e. There is other hcd that works with hub on stock examples such as rp2040 (pio and native), chipidea/ehci, ohci that you can take a look at. There is no docs but code is available.
